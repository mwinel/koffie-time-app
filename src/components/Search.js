import React, { Component } from "react";

class Search extends Component {
  render() {
    return (
      <div className="input-group mb-3">
        <input type="text" className="form-control" placeholder="Search posts..." />
      </div>
    );
  }
}

export default Search;
