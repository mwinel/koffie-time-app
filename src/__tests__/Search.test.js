import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import Search from "../components/Search";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders search component", () => {
  act(() => {
    render(<Search />, container);
  });
  expect(container.innerHTML).toBeDefined();
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
