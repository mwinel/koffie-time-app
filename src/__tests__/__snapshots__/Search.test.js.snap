// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search component 1`] = `"<div class=\\"input-group mb-3\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Search posts...\\"></div>"`;
